src/environmentd/tests/auth.rs

```diff
@@ -876,6 +960,24 @@ fn test_auth() {
         "TlsMode::Require, MzCloud",
         &server,
         &[
+            TestCase::Ws {
+                user: frontegg_user,
+                password: frontegg_password,
+                configure: Box::new(|b| Ok(b.set_verify(SslVerifyMode::NONE))),
+                assert: Assert::Success,
+            },
+            TestCase::Ws {
+                user: "bad user",
+                password: frontegg_password,
+                configure: Box::new(|b| Ok(b.set_verify(SslVerifyMode::NONE))),
+                assert: Assert::Err(Box::new(|code, message| {
+                    assert_eq!(code, CloseCode::Protocol);
+                    assert_eq!(
+                        message,
+                        "email in access token did not match the expected email"
```

Comment:
We apparently leak the error string. At one point this was logged and a more generic version was always returned to the user. We might consider bringing that back.